File: dfs.cpp

#include <iostream>
#include<vector>
#include<stack>
using namespace std;
vector<int> v[6];
void insert(int s,int d)
{
    v[s].push_back(d);
    
}
void dfs(int val)
{
    stack<int> st;
    bool visited[6];
    st.push(val);
    visited[val]=true;
    while(!st.empty())
    {
        int store=st.top();
        cout<<store<<" ";
        st.pop();
        for(int i=0;i<v[store].size();i++)
        {
            if(!visited[v[store][i]])
            {
              visited[v[store][i]]=true; 
            }
            st.push(v[store][i]);
            
            
            
        }
        
    }
    
    

}
int main()
{
    insert(0,1);
    insert(0,2);
    insert(0,3);
    insert(1,4);
    insert(2,5);
    //for(auto it:v[0])
    //cout<<it<<" ";
    dfs(0);
    
    return 0;


}

File: dfs.py
visited=set()
def dfs(graph,visited,src):
    if src not in visited:
        visited.add(src)
        print(src)
        for i in graph[src]:
            dfs(graph,visited,i)
            
        
graph = {
    'A' : ['B','C'],
    'B' : ['D', 'E'],
    'C' : ['F'],
    'D' : [],
    'E' : ['F'],
    'F' : []
}


dfs(graph,visited,'A')
